### ISSUE TITLE
missing human DEFB112 annotation 

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-annotation#2081]
Q30KQ8
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q30KQ8
has no process annotation in AmiGO, but has an InterPRo and UniProtKB-KW mapping

in QuickGO has

https://www.ebi.ac.uk/QuickGO/annotations?geneProductId=Q30KQ8
UniProtKB:Q30KQ8 | DEFB112 | involved_in | GO:0006952    defense response | ECO:0000322 IEA | GO_REF:0000037 | UniProtKB-KW:KW-0211 | 9606 | UniProt |  
UniProtKB:Q30KQ8 | DEFB112 | involved_in | GO:0042742    defense response to bacterium | ECO:0000322 IEA | GO_REF:0000037 | UniProtKB-KW:KW-0044 | 9606 | UniProt |  
UniProtKB:Q30KQ8 | DEFB112 | involved_in | GO:0045087    innate immune response | ECO:0000256 IEA | GO_REF:0000002 | InterPro:IPR025933 | 9606 | InterPro

none of these annotations make it into the GO database , why is that?
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q30KQ8

**Comment by ValWood** [comment]
I found them here
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:A0A287BBL9

but how does this get into AmiGO since it isn't  the entry in the reference proteome set? It really is IMPOSSIBLE to use this data ....we have multiple problems every time we attempt to do any analysis....it's very frustrating-  I can't even think of a good workaround for this. 
We have been trying to  finish this  analysis for a year now.....

**Comment by ValWood** [comment]
In fact when I go to the Gorilla antry I see the  PAINT annotation:
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q30KL5
but that isn't on the human either?

**Comment by ValWood** [comment]
Added to
Representing complete proteomes in GO
agenda item
http://wiki.geneontology.org/index.php?title=2018_Montreal_GOC_Meeting_Agenda&action=edit&section=16

**Comment by ValWood** [comment]
@pgaudet  who could troubleshoot this. It seems that it could be either a reference proteome issue, or a GO pipeline issue?

**Comment by pgaudet** [comment]
@kltm Any idea ?

**Comment by kltm** [comment]
What's the upstream source file that we should be poking around in? We can then see if that line has a rule violation.

**Comment by ValWood** [comment]
I don't think it's a rule violation. 
there are lots of annotations missing SP:KW and InterPRO
https://www.ebi.ac.uk/QuickGO/annotations?geneProductId=Q30KQ8

**Comment by kltm** [comment]
"Upgrades" and removals can be rule violations (or may be mapped onto a rule in the future); first we have to see if things exist in our upstream. If they do not exist in our upstream, then it is a policy issue: do we need to ingest more/different files.

**Comment by ValWood** [comment]
OK, but I can't see how all of the above can be rule violation. Some maybe, but not

UniProtKB:Q30KQ8 | DEFB112 | involved_in | GO:0045087    innate immune response | ECO:0000256 IEA | GO_REF:0000002 | InterPro:IPR025933 | 9606 | InterPro

looks kosher. 
It's from a pipeline so syntax errors unlikely. No taxon violations. No "do not annotate". What else is there?

**Comment by ValWood** [comment]
Where can we look up an individual entry to see if there are problems? Is that possible?

**Comment by kltm** [comment]
What is the origin file?

**Comment by ValWood** [comment]
QuickGO
https://www.ebi.ac.uk/QuickGO/annotations?geneProductId=Q30KQ8

AmiGO
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q30KQ8

**Comment by kltm** [comment]
For example if I check our upstream goa files for annotations with GO:0005576 and  Q30KQ8: 
bbop@wok:/home/skyhook/release/products/annotations$ zgrep GO:0005576 goa_*-src.gaf.gz | grep Q30KQ8

```
goa_human-src.gaf.gz:UniProtKB	Q30KQ8	DEFB112		GO:0005576	GO_REF:0000039	IEA	UniProtKB-SubCell:SL-0243	C	Beta-defensin 112	DEFB112|DEFB12	protein	taxon:9606	20180714	UniProt		
goa_uniprot_all-src.gaf.gz:UniProtKB	Q30KQ8	DEFB112		GO:0005576	GO_REF:0000002	IEA	InterPro:IPR025933	C	Beta-defensin 112	DEFB112|DEFB12	protein	taxon:9606	20180714	InterPro		
goa_uniprot_all-src.gaf.gz:UniProtKB	Q30KQ8	DEFB112		GO:0005576	GO_REF:0000037	IEA	UniProtKB-KW:KW-0964	C	Beta-defensin 112	DEFB112|DEFB12	protein	taxon:9606	20180714	UniProt		
goa_uniprot_all-src.gaf.gz:UniProtKB	Q30KQ8	DEFB112		GO:0005576	GO_REF:0000039	IEA	UniProtKB-SubCell:SL-0243	C	Beta-defensin 112	DEFB112|DEFB12	protein	taxon:9606	20180714	UniProt		
```

If I run the same check on our "product" files:

```
bbop@wok:/home/skyhook/release/annotations$ zgrep GO:0005576 goa_*.gaf.gz | grep Q30KQ8
goa_human.gaf.gz:UniProtKB	Q30KQ8	DEFB112		GO:0005576	GO_REF:0000039	IEA	UniProtKB-SubCell:SL-0243	C	Beta-defensin 112	DEFB112|DEFB12	protein	taxon:9606	20180714	UniProt		
```

So while there were more annotations at the beginning of the pipeline, all of the human ones were dropped from the "all" GAF, likely due to some rule or policy.

@dougli1sqrd Would you happen to know what that filter was?

**Comment by cmungall** [comment]
Eric, goa_uniprot_all is a red herring here. But in answer to your question, we want there to be one file for human.

Can we try and stick to one ID and one species. Otherwise things get confusing fast due to the fact that sometimes things change upstream on a daily basis.

The original question was about Q30KQ8, so I will answer that (spoiler: there is no mystery here)

This is what we received from the upstream (goa) for the last release, which was built on 2018-09-05


```
$ curl -L -s http://release.geneontology.org/2018-09-05/products/annotations/goa_human-src.gaf.gz | gzip -dc | grep Q30KQ8
UniProtKB       Q30KQ8  DEFB112         GO:0005576      GO_REF:0000039  IEA     UniProtKB-SubCell:SL-0243       C       Beta-defensin 112DEFB112|DEFB12   protein taxon:9606      20180714        UniProt 
UniProtKB       Q30KQ8  DEFB112         GO:0009986      PAINT_REF:15001 IBA     PANTHER:PTN001430319    C       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20140829        GO_Central      
UniProtKB       Q30KQ8  DEFB112         GO:0042742      PAINT_REF:15001 IBA     PANTHER:PTN001430319    P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20140829        GO_Central      
UniProtKB       Q30KQ8  DEFB112         GO:0045087      PAINT_REF:15001 IBA     PANTHER:PTN001430319    P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20140829        GO_Central      
```

One IEA and 3 IBAs. 

As we all know, the IBAs always get filtered out at source, and the paint annotations for this gene product are merged/injected in.

Let's look at what would be merged in:

```
$ curl -L -s http://release.geneontology.org/2018-09-05/products/annotations/paint_goa_human-src.gaf.gz | gzip -dc | grep Q30KQ8
```
(empty file)

I am assuming this is correct, but if you believe there should be paint annotations for Q30KQ8 then a ticket should be filed for this.

This explains why we see one annotation in AmiGO on http://amigo.geneontology.org/amigo/gene_product/UniProtKB:Q30KQ8

So everything is working precisely as intended right now.

But why is QuickGO showing different results than AmiGO today? That's also easily explained.

Now let's look at what goa is providing **right now**:

```
$ curl -L -s ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/HUMAN/goa_human.gaf.gz | gzip -dc | grep Q30KQ8
UniProtKB       Q30KQ8  DEFB112         GO:0005576      GO_REF:0000039  IEA     UniProtKB-SubCell:SL-0243       C       Beta-defensin 112DEFB112|DEFB12   protein taxon:9606      20180908        UniProt 
UniProtKB       Q30KQ8  DEFB112         GO:0042742      GO_REF:0000037  IEA     UniProtKB-KW:KW-0044    P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20180908        UniProt 
UniProtKB       Q30KQ8  DEFB112         GO:0045087      GO_REF:0000002  IEA     InterPro:IPR025933      P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20180908        InterPro
```

Compare this to the annotations for 2018-09-05, which I showed above.

Please note the dates on these annotations. They are *after* the current GO Central release date, so it is not a mystery that there are differences.

Let's have a look at what's in QuickGO right now:

https://www.ebi.ac.uk/QuickGO/annotations?geneProductId=Q30KQ8
![image](https://user-images.githubusercontent.com/50745/46043759-a5f62200-c0cd-11e8-9387-bc6e675f118f.png)

I used the "customize" button on QuickGO to show the date. I recommend everyone reporting do this (perhaps @tonysawfordebi can make this a default).

As you can see from my screenshot, QuickGO is actually __ahead__ of what is in the GOA release files (released 2018-09-10), and includes even _more_ annotations than when Val first reported this. This kind of rapid change makes explaining what is happening here more easily (@tonysawfordebi  - is there a reason QuickGO is ahead of the GOA releases? Have you thought of keeping these in sync).

To summarize

 * GOA and GO Central have different release cycles, so we expect differences. We can talk about aligning these cycles in Montreal
 * QuickGO is ahead of the GOA release files, so we can expect 3-way differences between AmiGO/GO Central vs GOA files vs QuickGO
 * I did not check on whether PAINT annotations for Q30KQ8 should be present. If someone believes that PANTHER should have immune annotations for this protein (seems reasonable) then I suggest opening a fresh ticket

Some additional tips:

 * use the customize feature in QuickGO to show dates and report them here (or first check them to see if your issue can be explained by the dates)

**Comment by ValWood** [comment]
Yep,
UniProtKB       Q30KQ8  DEFB112         GO:0042742      PAINT_REF:15001 IBA     PANTHER:PTN001430319    P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20140829        GO_Central      
UniProtKB       Q30KQ8  DEFB112         GO:0045087      PAINT_REF:15001 IBA     PANTHER:PTN001430319    P       Beta-defensin 112       DEFB112|DEFB12    protein taxon:9606      20140829        GO_Central      

seemed sensible and have disappeared.

Sorry about this one. It isn't possible for me to see the dates in QuickGO,  and I assumed the KW was old as I didn't think new mappings were being added ....

**Comment by ValWood** [comment]
Actually I'll reopen this one and assign to @pgaudet  since this is the annotation tracker

**Comment by ValWood** [comment]
closing this one I have enough examples
